test/utils/test_sparse.py

```diff
@@ -0,0 +1,38 @@
+"""Tests from_sparse utility function."""
+
+import numpy as np
+import pytest
+import torch
+from scipy import sparse
+
+from topomodelx.utils.sparse import from_sparse
+
+
+def test_from_sparse():
+    """Tests from_sparse matches sparse -> dense -> sparse."""
+    # test values matche
+    test_matrix = sparse._csc.csc_matrix(np.random.rand(100, 100))
+    a = torch.from_numpy(test_matrix.todense()).to_sparse()
+    b = from_sparse(test_matrix)
+
+    assert torch.equal(
+        a.to_dense().type(torch.float32), b.to_dense().type(torch.float32)
+    )
```

Comment:
You can use `torch.testing.assert_close` here. You don't need to convert to dense and to float.

Reply:
I changed to `torch.testing.assert_close` but I needed the conversion to dense otherwise it fails due to mismatch in index representation:
```
print(np.allclose(b.to_dense().numpy(), test_matrix.todense())) #passes
print(np.allclose(a.to_dense().numpy(), test_matrix.todense())) #passes
torch.testing.assert_close(a.to_dense(), b.to_dense()) #passes
torch.testing.assert_close(a, b) #fails
```